org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/SocketServerLauncher.xtend

```diff
@@ -0,0 +1,100 @@
+/*******************************************************************************
+ * Copyright (c) 2019 TypeFox GmbH (http://www.typefox.io) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+ 
+package org.eclipse.xtext.ide.server
+
+import com.google.inject.Guice
+import java.io.PrintWriter
+import java.net.InetSocketAddress
+import java.nio.channels.AsynchronousServerSocketChannel
+import java.nio.channels.Channels
+import org.eclipse.lsp4j.jsonrpc.Launcher
+import org.eclipse.lsp4j.services.LanguageClient
+import org.eclipse.xtext.ide.server.LanguageServerImpl
+import org.eclipse.xtext.ide.server.ServerModule
+import org.eclipse.xtext.util.internal.Log
+
+/**
+ * Allows to launch a language server vie a server socket.
+ * 
+ * Useful for debugging live language servers.
+ * 
+ * @author Jan K&ouml;hnlein - Initial contribution and API
+ * @since 2.18
+ */
+@Log
+class SocketServerLauncher {
+
+	static val HOST = '-host'
```

Comment:
maybe should be made public analogous to ServerLauncher

Reply:
I cannot think of any usecase where I really want to access that from the outside.

Reply:
i assume the usecase is subclassing (am not sure why they are public in server launcher)